core/src/ibc/channel_04/manager.rs

```diff
@@ -40,4 +44,61 @@ impl<'a> Manager<'a> {
             ctx,
         }
     }
+
+    pub fn chan_open_init(
+        &mut self,
+        order: ChannelOrder,
+        connection: Identifier,
+        channel_identifier: Identifier,
+        counterparty_channel_identifier: Identifier,
+        version: String,
+    ) -> Result<Identifier, String> {
+        let kv_store = self.ctx.get_kv_store_mut();
+
+        // It is ok to be in any state.
```

Comment:
Could you add a detailed description of this?
"It is ok to be in any state" does not give the intention to the reader.

The ICS spec is saying "optimistic" at the same point.
After reading it, I can understand the behavior. Channel can be initiated while the connection is establishing.

So, when you write a comment, please explain the intention of code, not the behavior of code.